### PR TITLE
[KEYCLOAK-11435] Log deprecation warning for usage of non-native Prom…

### PR DESCRIPTION
**Pull request by jonkoops**

### CODE REVIEW

File: adapters/oidc/js/src/main/resources/keycloak.js [hunk 1]
```diff
@@ -110,8 +110,9 @@
                 }
 
                 if (initOptions.promiseType === 'native') {
-                    kc.useNativePromise = typeof Promise === "function";
```
Comment (jonkoops) [inline adapters/oidc/js/src/main/resources/keycloak.js]: I've removed the feature detection here as it doesn't make sense to fall back to the legacy API if the consumer of the API is expecting a different interface. Better to fail hard constructing a promise then to pretend we can fix things.

File: adapters/oidc/js/src/main/resources/keycloak.d.ts [hunk 2]
```diff
@@ -271,15 +279,14 @@ declare namespace Keycloak {
 	 * Conditional CompatPromise type in order to support
 	 * both legacy promises and native promises as return types.
 	 */
-	type PromiseType = KeycloakPromiseType | undefined;
-	type CompatPromise<TPromiseType extends PromiseType, TSuccess, TError> =
```
Comment (jonkoops) [inline adapters/oidc/js/src/main/resources/keycloak.d.ts]: I tried to work with the types that were provided initially here but when creating a Keycloak instance I always got the native promise version even if the generic was never passed properly. I assume this was a bug which should now be fixed.

So just to clarify:

```
const instance = Keycloak(); // Typescript assumes this is a 'legacy' promise all methods return .success() and .error()
const instance = Keycloak<'native'>(); // Typescript assumes this is a native promise all methods return standard promises
```

File: adapters/oidc/js/src/main/resources/keycloak.d.ts [hunk 3]
```diff
@@ -271,15 +279,14 @@ declare namespace Keycloak {
 	 * Conditional CompatPromise type in order to support
 	 * both legacy promises and native promises as return types.
 	 */
-	type PromiseType = KeycloakPromiseType | undefined;
```
Comment (jonkoops) [inline adapters/oidc/js/src/main/resources/keycloak.d.ts]: This extra type has been removed as `'legacy'` is now passed everywhere explicitly.

**Review by mhajas — APPROVED** [review]
I like this change. It looks good and reasonable to me. However, I don't feel very confident in TypeScript. @stianst could you please look at this as well?

**Review by mhajas — APPROVED** [review]

**Review by ssilvert — APPROVED** [review]
This looks good and I'm approving the changes.

The only thing that worries me a bit is the use of generic parameter defaults.  This was introduced in TypeScript 2.3, which was released on 27 April 2017.  So customers who are stuck on TypeScript 2.2 or earlier will need a workaround.  I don't know if many customers would be affected.

**Pull request merged**